### PR TITLE
Add pushing and pulling gittuf to dogfooding docs

### PR DESCRIPTION
**Pull request by neilnaveen**
Add pushing and pulling gittuf to dogfooding docs

### CODE REVIEW

File: docs/dogfood.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ as gittuf itself may have breaking changes in the coming months. After gittuf
 reaches v1, we expect to reset the policy and start over with a formal root
 signing. We envision dogfooding to happen in several phases.
 
+**NOTE: when using gittuf with any server, to push and pull gittuf specific refs (e.g RSL, Policy, and Attestations), `git push remote-name refs/gittuf/*` and `git fetch remote-name refs/gittuf/*:refs/gittuf/*` must be used instead of gittuf commands for the v.0.4.0 relese**
```
Comment (adityasaky) [inline docs/dogfood.md:12]: This sort of comes out of nowhere right now. I'd make this a whole section below titled "Verifying gittuf using gittuf", which would include instructions for using either the gittuf CLI to clone the repo and verify, or the git CLI.

Comment (adityasaky) [reply docs/dogfood.md:12]: Also, please wrap at 80 cols!

Comment (neilnaveen) [reply docs/dogfood.md]: Wrapped, and I have created a new section

**Review by adityasaky — CHANGES_REQUESTED** [review]

File: docs/dogfood.md [hunk 2]
```diff
@@ -35,3 +35,23 @@ primarily offline signatures. This can, as before, only be achieved with further
...
+
+### Verifying `gittuf` using `gittuf`
+
+Before verifying using gittuf, we must first clone the repo. If you clone with
+`gittuf clone https://github.com/gittuf/gittuf`, all gittuf metadata will be
+pulled down. If instead you perform a regular `git clone`, or `git pull`, you
+will have to use `git fetch remote-name refs/gittuf/*:refs/gittuf/*`, to pull
+in gittuf metadata
+
+After this you should be able to perform verification. An easy verification
```
[the remark this answers is not in the document]
Comment (neilnaveen) [reply docs/dogfood.md]: Fixed

File: docs/dogfood.md [hunk 3]
```diff
@@ -35,3 +35,23 @@ primarily offline signatures. This can, as before, only be achieved with further
 usability improvements. In this final phase, we hope to essentially have worked
 out the kinks with using gittuf actively so that we can proceed with a stable
 release.
+
+### Verifying `gittuf` using `gittuf`
+
+Before verifying using gittuf, we must first clone the repo. If you clone with
```
[the remark this answers is not in the document]
Comment (neilnaveen) [reply docs/dogfood.md]: Fixed

File: docs/dogfood.md [hunk 4]
```diff
@@ -35,3 +35,23 @@ primarily offline signatures. This can, as before, only be achieved with further
 usability improvements. In this final phase, we hope to essentially have worked
 out the kinks with using gittuf actively so that we can proceed with a stable
 release.
+
+### Verifying `gittuf` using `gittuf`
```
[the remark this answers is not in the document]
Comment (neilnaveen) [reply docs/dogfood.md]: Fixed

**Review by adityasaky — APPROVED** [review]

**Pull request merged**